src/App.jsx

```diff
@@ -14,45 +17,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = goods.map((good, index) => ({
+    id: index + 1,
+    name: good,
+  }));
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      const { name: good1 } = a;
+      const { name: good2 } = b;
+
+      switch (sortField) {
+        case SORT_FIELD_ABC:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          throw new Error(`Invalid sortField: ${sortField}`);
+      }
+    });
+  }
+
+  return isReversed
+    ? preparedGoods.reverse()
+    : preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
+  const resetBtnFilter = sortField !== '' || isReversed;
```

Comment:
```suggestion
  const isResetActive = sortField !== '' || isReversed;
```
The name should specify that it's a boolean value
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/6de832e6-8903-4c12-b54a-52aa84368910)